docs/lesson2.md

```diff
@@ -0,0 +1,206 @@
+## Chomping
+
+A more low-level operation is "chomping". "chomping" here means consuming characters, moving along in the input string, but not actually doing anything with the consumed characters. The chomping parsers all have type `Parser ()`, not e.g. `Parser String` like you might expect. We'll get back to this later.
+
+The `chompIf` parser will chomp exactly one character if the predicate (a `Char -> Bool` function) is true for the next character of the input.
+
+```elm
+chompIf : (Char -> Bool) -> Parser () 
+```
+
+The `chompWhile` parser will continually chomp characters if the condition remains true.
+
+```elm
+chompWhile : (Char -> Bool) -> Parser () 
+```
+
+### Examples
+
+Chomping a positive integer. 
```

Comment:
Pedagogy is all about telling lies and refining them later on.

The `chompInteger` also accepts input like `00001234`, i.e. a an non-empty sequence of digits possible possible starting with a `0`. 

Do we want to address this fact?

Reply:
Hi Folkert, you've done a really nice job of explaining some difficult topics.  

- I especially like the way you introduced `loop` by first remarking on how you parse something twice.   

- We may want to talk about `andThen` in Lesson 1 (or somewhere)

- Ha ha!  Which joke?

- Do we need to talk about the type of `Parser.problem`?

- It would be good to also have some exercises to go with this write-up